code/datums/storage/subtypes/rped.dm

```diff
@@ -9,7 +9,77 @@
 	max_specific_storage = WEIGHT_CLASS_NORMAL
 	numerical_stacking = TRUE
 
+	var/list/allowed_mat_types=list(
+		/obj/item/stack/sheet/glass,
+		/obj/item/stack/sheet/plasteel,
+		/obj/item/stack/cable_coil
+	)
+
+	var/list/allowed_bluespace_types=list(
+		/obj/item/stack/ore/bluespace_crystal,
+		/obj/item/stack/ore/bluespace_crystal/refined,
+		/obj/item/stack/ore/bluespace_crystal/artificial,
+		/obj/item/stack/sheet/bluespace_crystal
+	)
+
+
 /datum/storage/rped/can_insert(obj/item/to_insert, mob/user, messages = TRUE, force = FALSE)
 	. = ..()
-	if(!to_insert.get_part_rating())
+
+	///we check how much of glass,plasteel & cable the user can insert
+	if(isstack(to_insert))
+		///user tried to insert invalid stacktype
+		if(!is_type_in_list(to_insert,allowed_mat_types) && !is_type_in_list(to_insert,allowed_bluespace_types))
+			return FALSE
+
+		var/obj/item/stack/the_stack=to_insert
+		//how much of the stack is the user trying to insert
+		var/insert_amount=the_stack.amount
+		//how much of this stack type is currently in the stack
+		var/present_amount=0
+		//what is the maximum allowed amount for this stack type in storage
+		var/max_amount=30
+		//how much space is available
+		var/available=0
+		///stacks type
+		var/obj/item/stack/things
+
+		//not a real location so dont bother
+		var/obj/item/resolve_location = real_location?.resolve()
+		if(!resolve_location)
+			return FALSE
+
+		///we try to count & limit how much the user can insert of each type to prevent them from using it as an normal storage medium
+		for(var/obj/item/thing in resolve_location.contents)
+			///try convert to stack else skip loop as we are only intrested in counting stacks
+			if(isstack(thing))
+				things=thing
+			else
+				continue
+
+			///count how many of this stacktype is already in storage. One type of bluespace crystal takes space for all other bluespace types as well
+			if(is_type_in_list(to_insert,allowed_bluespace_types))
+				if(is_type_in_list(things,allowed_bluespace_types))
+					present_amount+=things.amount
+			else if(istype(to_insert,things.type))
+				present_amount=things.amount
+				break
+
+
+		///no more storage for this specific stack type
+		if(max_amount-present_amount==0)
+			to_chat(usr,span_alert("No more [to_insert.name] can be added!"))
```

Comment:
Does any other `can_insert` send to_chats?

Reply:
Also use `user`, not `usr`.

Reply:
Only this one display chat, your right that does seem off a bit. Ill remove the chat. Renamed usr->user